qiskit/algorithms/minimum_eigen_solvers/qaoa.py

```diff
@@ -125,14 +125,12 @@ def __init__(
             quantum_instance=quantum_instance,
         )
 
-    def _check_operator(self, operator: OperatorBase) -> OperatorBase:
+    def _check_operator_varform(self, operator: OperatorBase) -> OperatorBase:
```

Comment:
Would this, and the one in VQE - be better named `_check_operator_ansatz` or something like that - mainly lose/replace the 'varform' bit which is named now based really on what we used to have and may not make sense coming new to the code